### PR TITLE
Update tools

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR presents a major update to the tools. It collects all latests developments (e.g., for C# and Manuscript) and brings:

* drops Python 2 support
* switches from `os.path` to `pathlib` (simplifying several things)
* sorting imports (PEP 8)
* formatting (PEP 8) and adding docstrings
* adds a `requirements.txt`

Most important for Verovio context:
For retrieving `tei:desc` it switches from `text()` to `string()` importing the complete description instead only the first chuck to a subelement.

### CODE REVIEW

File: tools/langs/cplusplus.py [hunk 1]
```diff
@@ -231,15 +228,14 @@ def __create_mixin_classes(schema, outdir):
...
+            'moduleNameCaps': "{0}MIXIN".format((module.upper()).replace("-", "_")),
             'elements': classes.strip()
         }
         if "std::string" in classes:
             tplvars["includes"] = "#include <string>"
         fullout = CLASSES_HEAD_TEMPLATE.format(**tplvars)
-        fmh = open(os.path.join(outdir, "{0}mixins.h".format(module.lower())), 'w')
-        fmh.write(fullout)
-        fmh.close()
+        fmh = Path(outdir, "{0}mixins.h".format(module.lower()))
```
Comment (ahankinson) [inline tools/langs/cplusplus.py]: Should we take the further opportunity to switch to `f` strings?
```suggestion
        fmh = Path(outdir, f"{module.lower()}mixins.h")
```

File: tools/langs/cplusplus.py [hunk 2]
```diff
@@ -324,24 +319,27 @@ def __create_element_classes(schema, outdir):
                         if len(sda.split("|")) > 1:
                             ns, sda = sda.split("|")
 
-                        docstr = __get_docstr(schema.getattdocs(sda), indent=8)
+                        docstr = __get_docstr(
```
Comment (ahankinson) [inline tools/langs/cplusplus.py:322]: I'm not sure why this line is indented. It's very unlikely that it went over 80 chars?

File: tools/langs/cplusplus.py [hunk 3]
```diff
@@ -324,24 +319,27 @@ def __create_element_classes(schema, outdir):
...
+                            schema.get_att_desc(sda), indent=8)
                         methstr = {
                             "attNameUpper": schema.cc(schema.strpatt(sda)),
                             "attNameLower": sda,
                             "attNameLowerJoined": schema.strpdot(sda),
                             "documentation": docstr
                         }
-                        attribute_methods += METHODS_HEADER_TEMPLATE.format(**methstr)
+                        attribute_methods += METHODS_HEADER_TEMPLATE.format(
```
Comment (ahankinson) [inline tools/langs/cplusplus.py:330]: Indentation?

File: tools/langs/cplusplus.py (hunk 2 above)
Comment (rettinghaus) [reply tools/langs/cplusplus.py:322]: It's 82 chars.

File: tools/langs/cplusplus.py (hunk 3 above)
Comment (rettinghaus) [reply tools/langs/cplusplus.py:330]: 87 chars

File: tools/langs/__init__.py [hunk 4]
```diff
@@ -1 +1 @@
-AVAILABLE_LANGS = ["C++", "Python", "ManuScript", "Java"]
\ No newline at end of file
+AVAILABLE_LANGS = ("cpp", "csharp", "java", "manuscript", "python", "vrv", "vdoc")
```
Comment (lpugin) [inline tools/langs/__init__.py:1]: What is vdoc? The html_vrv?

Comment (rettinghaus) [reply tools/langs/__init__.py:1]: Yes, this is what I found in `parseschema`.

**Review by lpugin — APPROVED** [review]

**Pull request merged**